tests/behaviour/util.rs

```diff
@@ -19,8 +19,166 @@
+                .await
+        }
+        _ => unreachable!("Unexpected Concept type: {answer:?}"),
+    };
+    match res {
+        Ok(_) => {
+            let equals = res
+                .unwrap()
+                .into_iter()
+                .filter(|attr| attr.type_.label == identifiers[0])
+                .collect::<Vec<_>>()
+                .first()
+                .and_then(|attr| Some(value_equals_str(&attr.value, identifiers[1])));
+            match equals {
+                Some(val) => val,
+                None => false,
+            }
+        }
+        Err(_) => false,
+    }
+}
+
+fn labels_equal(expected_label: &str, answer: &Concept) -> bool {
+    let mut binding = String::new();
+    let label = match answer {
+        Concept::EntityType(type_) => &type_.label,
+        Concept::RoleType(RoleType { label, .. }) => {
+            binding = format!("{label}");
+            &binding
+        }
+        Concept::Entity(Entity { type_: EntityType { label, .. }, .. }) => label,
+        Concept::Relation(Relation { type_: RelationType { label, .. }, .. }) => label,
+        Concept::RootThingType(_) => {
+            binding = String::from("thing");
+            &binding
+        }
+        Concept::RelationType(RelationType { label, .. }) => label,
+        Concept::AttributeType(AttributeType { label, .. }) => label,
+        Concept::Attribute(Attribute { type_: AttributeType { label, .. }, .. }) => label,
+    };
+    label == expected_label
+}
+
+fn values_equal(expected_label_and_value: &str, answer: &Concept) -> bool {
+    let identifiers: Vec<&str> = expected_label_and_value.splitn(2, ":").collect();
+    assert_eq!(identifiers.len(), 2, "Unexpected table cell format: {expected_label_and_value}.");
+    match answer {
+        Concept::Attribute(Attribute { value, .. }) => value_equals_str(value, identifiers[1]),
+        _ => false,
+    }
+}
+
+fn value_equals_str(value: &Value, expected: &str) -> bool {
+    match value {
+        Value::String(val) => val == expected,
+        Value::Long(val) => expected.parse::<i64>().and_then(|expected| Ok(expected.eq(val))).unwrap_or_else(|_| false),
+        Value::Double(val) => expected
+            .parse::<f64>()
+            .and_then(|expected| Ok(equals_approximate(expected, *val)))
+            .unwrap_or_else(|_| false),
+        Value::Boolean(val) => {
+            expected.parse::<bool>().and_then(|expected| Ok(expected.eq(val))).unwrap_or_else(|_| false)
+        }
+        Value::DateTime(val) => format_datetime(val) == expected,
+    }
+}
+
+fn get_iid(concept: &Concept) -> String {
+    let iid = match concept {
+        Concept::Entity(Entity { iid, .. }) => iid,
+        Concept::Attribute(Attribute { iid, .. }) => iid,
+        Concept::Relation(Relation { iid, .. }) => iid,
+        _ => unreachable!("Unexpected Concept type: {concept:?}"),
+    };
+    format!("0x{iid}")
+}
```

Comment:
This should be moved after `apply_query_template()`. We generally try to keep helpers right after the functions that use them.